### PR TITLE
admin_fine_logger: replace log level int with nice string

### PR DESCRIPTION
**Pull request by botengyao**
Before:
```
active loggers:
  source/server/admin/logs_handler.cc: 3
  source/common/common/logger.cc: 3
```

After:
```
active loggers:
  source/server/admin/logs_handler.cc: warn
  source/common/common/logger.cc: warn
```

admin test:

Screenshot 2024-06-20 at 8 58 37 AM

Commit Message:
Additional Description:
Risk Level: low
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: source/common/common/fine_grain_logger.cc [hunk 1]
```diff
@@ -85,7 +85,8 @@ std::string FineGrainLogContext::listFineGrainLoggers() ABSL_LOCKS_EXCLUDED(fine
   absl::ReaderMutexLock l(&fine_grain_log_lock_);
   std::string info =
       absl::StrJoin(*fine_grain_log_map_, "\n", [](std::string* out, const auto& log_pair) {
-        absl::StrAppend(out, "  ", log_pair.first, ": ", log_pair.second->level());
+        absl::StrAppend(out, "  ", log_pair.first, ": ",
+                        fmt::to_string(spdlog::level::to_string_view(log_pair.second->level())));
```
Comment (jmarantz) [inline source/common/common/fine_grain_logger.cc:89]: seems like we should be able to convert a spdlog string_view to an absl::string_view without deep-copy/allocation to a string.

**Review by jmarantz — COMMENTED** [review]
this looks great; modulo one nit. We should definitely have someone outside Google review.

Maybe @KBaichoo plus @ggreenway for senior approver. I don't think this is that controversial but maybe someone out there is depending on the existing format in scripts?

File: source/common/common/fine_grain_logger.cc (hunk 1 above)
Comment (botengyao) [reply source/common/common/fine_grain_logger.cc]: thanks, done!

**Review by jmarantz — APPROVED** [review]

**Review by jmarantz — APPROVED** [review]

**Review by KBaichoo — APPROVED** [review]
Thanks you, this makes sense to me.

I think because the fine grain logger is not on by default and I don't think we provide stability guarantees here for the logging output this should be okay to change.

**Review by ggreenway — APPROVED** [review]

**Pull request merged**